src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,120 @@
+package parser;
+
+import task.TaskList;
+
+/**
+ * The {@code Parser} class interprets user input and translates it into commands that modify the {@code TaskList}.
+ * It validates the input format before performing actions like marking, unmarking, deleting tasks, or adding new tasks.
+ */
+public class Parser {
+
+    /**
+     * Parses the input message for marking a task as done and applies the command to the given {@code TaskList}.
+     *
+     * @param message The input message to be parsed (e.g., "mark 1").
+     * @param t The {@code TaskList} where the task will be marked as done.
+     * @throws BobException If the message format is invalid.
+     */
+    public void markTaskParser(String message, TaskList t) throws BobException {
```

Comment:
The formatting of the names of the methods could be somewhat misleading. This method in particular could sound like you are marking a task parser. Perhaps something like parseMarkTaskCommand instead?

I've noticed the same thing in the Ui class as well.